vital/types/point.h

```diff
@@ -69,8 +69,8 @@ class VITAL_EXPORT point
     m_covariance = v;
   }
 
-  T& operator[]( int i ) { return m_value[i]; }
-  const T& operator[]( int i ) const { return m_value[i]; }
+  T operator[]( size_t i ) { return m_value[i]; }
```

Comment:
As structured, these overloads don't make sense. It would be more sensible for the non-`const` to return a reference and the `const` to return `T` (not `const T`, which is almost never desired as a return type).

(Also, per the style guide, this should be `T const`, though the above makes that irrelevant.)